crates/aptos/src/node/local_testnet/processors.rs

```diff
@@ -0,0 +1,193 @@
+use tokio::sync::OnceCell;
+use tracing::info;
+
+static RUN_MIGRATIONS_ONCE: OnceCell<bool> = OnceCell::const_new();
+
+/// This struct is used to parse the command line arguments for the processors.
+#[derive(Debug, Parser)]
+pub struct ProcessorArgs {
+    /// The value of this flag determines which processors we will run if
+    /// --with-indexer-api is set. Note that some processors are not supported in the
+    /// local testnet (e.g. ANS). If you try to set those, an error will be thrown
+    /// immediately.
+    #[clap(
+        long,
+        value_enum,
+        default_values_t = vec![
+            ProcessorName::AccountTransactionsProcessor,
+            ProcessorName::CoinProcessor,
+            ProcessorName::DefaultProcessor,
+            ProcessorName::EventsProcessor,
+            ProcessorName::FungibleAssetProcessor,
+            ProcessorName::StakeProcessor,
+            ProcessorName::TokenProcessor,
+            ProcessorName::TokenV2Processor,
+            ProcessorName::UserTransactionProcessor,
+        ],
+        requires = "with_indexer_api"
+    )]
+    processors: Vec<ProcessorName>,
+}
+
+#[derive(Debug)]
+pub struct ProcessorManager {
+    config: GenericConfig<IndexerGrpcProcessorConfig>,
+    prerequisite_health_checkers: HashSet<HealthChecker>,
+}
+
+impl ProcessorManager {
+    fn new(
+        processor_name: &ProcessorName,
+        prerequisite_health_checkers: HashSet<HealthChecker>,
+        health_check_port: u16,
+        data_service_url: Url,
+        postgres_connection_string: String,
+    ) -> Result<Self> {
+        let processor_config = match processor_name {
+            ProcessorName::AccountTransactionsProcessor => {
+                ProcessorConfig::AccountTransactionsProcessor
+            },
+            ProcessorName::AnsProcessor => {
+                bail!("ANS processor is not supported in the local testnet")
+            },
+            ProcessorName::CoinProcessor => ProcessorConfig::CoinProcessor,
+            ProcessorName::DefaultProcessor => ProcessorConfig::DefaultProcessor,
+            ProcessorName::EventsProcessor => ProcessorConfig::EventsProcessor,
+            ProcessorName::FungibleAssetProcessor => ProcessorConfig::FungibleAssetProcessor,
+            ProcessorName::NftMetadataProcessor => {
+                bail!("NFT Metadata processor is not supported in the local testnet")
+            },
+            ProcessorName::StakeProcessor => ProcessorConfig::StakeProcessor,
+            ProcessorName::TokenProcessor => {
+                ProcessorConfig::TokenProcessor(TokenProcessorConfig {
+                    // This NFT points thing doesn't exist on local testnets.
+                    nft_points_contract: None,
+                })
+            },
+            ProcessorName::TokenV2Processor => ProcessorConfig::TokenV2Processor,
+            ProcessorName::UserTransactionProcessor => ProcessorConfig::UserTransactionProcessor,
+        };
```

Comment:
nit - Personally I would add the unsupported processor into a set, then check from the list/set.